### PR TITLE
:sparkles: use bot account

### LINKED ISSUE #6: You can use [`GetBundlePath`](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetBundlePath) for this.
You can use [`GetBundlePath`](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetBundlePath) for this.

_Originally posted by @hanzei in https://github.com/manland/mattermost-plugin-gitlab/pull/5_

When mattermost 5.10 is out

### LINKED ISSUE #9: Usa bot account
Mattermost 5.10 introduces bot accounts. A bot account should be created on startup and used. The setting `Username` an be removed.

### LINKED ISSUE #20: Update section 4. of Setup Guide to use BOT Account
The setup guide should to be updated to use the new BOT accounts functionality. This functionality will be exposed in the UI for 5.12 

Section 4. - step ii.
`Create a new Mattermost user, through the regular UI or the CLI with the username "gitlab"`

This will should be undated for 5.12 to instruct the user to navigate to the integrations section and create a new bot instead of a user account that functions as a BOT.

### PR DESCRIPTION
**Pull request by manland**
Huge thanks to @crspeller for [inspiration](https://github.com/mattermost/mattermost-plugin-github/tree/bot-accounts)

Cc @hanzei @DHaussermann 

Fix #9 
Fix #6
Fix #20

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "name": "GitLab",
     "description": "GitLab plugin for Mattermost.",
     "version": "0.2.0",
+    "min_server_version": "5.10.0",
```
Comment (hanzei) [inline plugin.json]: :tada: :rocket:

File: server/command.go [hunk 2]
```diff
@@ -47,7 +48,7 @@ func (p *Plugin) getCommandResponse(responseType, text string) *model.CommandRes
 		ResponseType: responseType,
 		Text:         text,
 		Username:     GITLAB_USERNAME,
-		IconURL:      p.getConfiguration().ProfileImageURL,
+		IconURL:      path.Join(*p.API.GetConfig().ServiceSettings.SiteURL, "plugins", manifest.Id, "assets", "profile.png"),
```
Comment (hanzei) [inline server/command.go]: You can use [`GetBundlePath`](https://developers.mattermost.com/extend/plugins/server/reference/) instead.

File: server/go.mod [hunk 3]
```diff
@@ -3,48 +3,16 @@ module github.com/manland/mattermost-plugin-gitlab/server
...
-	github.com/hashicorp/go-plugin v0.0.0-20180814222501-a4620f9913d1 // indirect
-	github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d // indirect
 	github.com/lib/pq v1.1.0 // indirect
-	github.com/mattermost/gorp v2.0.0+incompatible // indirect
-	github.com/mattermost/mattermost-server v5.10.0+incompatible
-	github.com/mattermost/viper v0.0.0-20181112161711-f99c30686b86 // indirect
-	github.com/mattn/go-sqlite3 v1.10.0 // indirect
-	github.com/mitchellh/go-testing-interface v1.0.0 // indirect
-	github.com/mitchellh/mapstructure v1.1.2 // indirect
+	github.com/mattermost/mattermost-server v0.0.0-20190509204005-3f3bc5c49c37
```
Comment (hanzei) [inline server/go.mod]: :tada:

File: server/plugin.go [hunk 4]
```diff
@@ -52,19 +55,35 @@ func (p *Plugin) OnActivate() error {
...
+
+	bundlePath, err := p.API.GetBundlePath()
 	if err != nil {
-		p.API.LogError("can't get user by username", "err", err.Error())
-		return fmt.Errorf("Unable to find user with configured username: %v", config.Username)
+		return errors.Wrap(err, "can't retreive bundle path")
 	}
 
-	p.BotUserID = user.Id
+	p.BotUserID = botID
```
Comment (hanzei) [inline server/plugin.go]: Nit: Move this in line 70.

File: server/command.go (hunk 2 above)
Comment (manland) [reply server/command.go]: Nop, GetBundlePath return an absolute path e.g. `/home/user/....` I want an http URL. Maybe a cool helper could be `GetHTTPBundlePath`!?

Comment (hanzei) [reply server/command.go]: :man_facepalming: Your correct.

Comment (hanzei) [reply server/command.go]: What would `GetHTTPBundlePath()` return? `ServiceSettings.SiteURL` + "plugins" + `pluginID`?

Comment (manland) [reply server/command.go]: Yes :100:

File: plugin.json (hunk 1 above)
Comment (hanzei) [reply plugin.json]: Please keep in mind that you have to bump this to 5.12 eventually.

**Pull request merged**